subsys/net/lib/rest_client/Kconfig

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig REST_CLIENT
+	bool "REST client library"
+	select HTTP_CLIENT
+
+if REST_CLIENT
+
+config REST_CLIENT_REQUEST_TIMEOUT
+	int "Rest request timeout, in seconds"
+	default 65
+	help
+	  Sets the timeout duration in seconds to wait for the response data.
+
+config REST_CLIENT_SCKT_SEND_TIMEOUT
+	int "Socket send timeout, in seconds"
+	default 60
+	help
+	  Sets the timeout duration in seconds to use when sending data.
+	  To disable, set the timeout duration to -1.
+
+config REST_CLIENT_SCKT_RECV_TIMEOUT
+	int "Socket receive timeout, in seconds"
+	default 60
+	help
+	  Sets the timeout duration in seconds to use when receiving data.
+	  To disable, set the timeout duration to -1.
+
+config REST_CLIENT_SCKT_TLS_SESSION_CACHE_IN_USE
+	bool "TLS session cache usage"
+	default y
+	help
+	  TLS session cache, disable or enable.
```

Comment:
I'm wondering if it would be better to have this in the runtime configuration. It could be desirable to enable this for some servers but not for others. Or it could be that some application/library wants to enable this, but some doesn't. Moving this to the client context would make it more flexible.

Reply:
I was thinking similarly but ended to this solution. After some testings, seems that TLS session cache is a very good feature for the usage like REST req/resps (usually towards the same server), and by this way we are pushing more the usage. Thus, I'd keep this as it is now and act later if needed.